index.js

```diff
@@ -5249,10 +5249,12 @@ function StartTSLClientConnection(tslClientId) {
 						UpdateSockets('tsl_clients');
 					});
 					tsl_clients[i].socket.on('close', function() {
-						logger(`TSL Client ${tslClientId} Connection Closed: ${tsl_clients[i].ip}:${tsl_clients[i].port}`, 'info-quiet');
-						tsl_clients[i].error = false;
-						tsl_clients[i].connected = false;
-						UpdateSockets('tsl_clients');
+						if (tsl_clients[i]) {
+							logger(`TSL Client ${tslClientId} Connection Closed: ${tsl_clients[i]?.ip}:${tsl_clients[i]?.port}`, 'info-quiet');
```

Comment:
I had to remove these question marks `tsl_clients[i]?.ip` to `tsl_clients[i].ip` for it to work

Reply:
Oops, should be fixed. You're probably using Node 13 or lower as optional chaining arrived with Node v14. I had to add the if condition because the server would crash when removing a disconnected TSL Client. If you want, I can revert this change as it is beyond the scope of this PR.

Reply:
I’m ok with it as long as we make sure people know they need to upgrade node. Companion requires 12.21 so generally that’s what I use for other stuff also.

Reply:
Sounds good, but with 97e9b0a we should be compatible again anyway. The question marks and the if condition were Kind of redundant, so I removed them again. 👍